src/System.IO.FileSystem.AccessControl/src/System/IO/FileSystemAclExtensions.netcoreapp.cs

```diff
@@ -24,5 +40,162 @@ public static void Create(this DirectoryInfo directoryInfo, DirectorySecurity di
+        /// <param name="share">One of the enumeration values for controlling the kind of access other FileStream objects can have to the same file.</param>
+        /// <param name="bufferSize">The number of bytes buffered for reads and writes to the file.</param>
+        /// <param name="options">One of the enumeration values that describes how to create or overwrite the file.</param>
+        /// <param name="fileSecurity">An object that determines the access control and audit security for the file.</param>
+        /// <returns>A file stream for the newly created file.</returns>
+        /// <exception cref="ArgumentNullException"><paramref name="fileInfo" /> or <paramref name="fileSecurity" /> is <see langword="null" />.</exception>
+        /// <exception cref="ArgumentOutOfRangeException"><paramref name="mode" /> or <paramref name="share" /> are out of their legal enum range.
+        ///-or-
+        /// <paramref name="options" /> is not a supported enum value.
+        ///-or-
+        /// <paramref name="bufferSize" /> is not a positive number.</exception>
+        /// <exception cref="UnauthorizedAccessException">Access to the path is denied.</exception>
+        /// <exception cref="ArgumentException">Invalid handle.</exception>
+        /// <exception cref="ObjectDisposedException">Cannot access a closed file.</exception>
+        /// <remarks><format type="text/markdown"><![CDATA[
+        ///## Remarks
+        ///This extension method was added to .NET Core to bring the functionality that was provided by the <xref:System.IO.FileStream.#ctor(System.String,System.IO.FileMode,System.Security.AccessControl.FileSystemRights,System.IO.FileShare,System.Int32,System.IO.FileOptions,System.Security.AccessControl.FileSecurity)> .NET Framework constructor.
+        /// ]]></format></remarks>
+        public static FileStream Create(this FileInfo fileInfo, FileMode mode, FileSystemRights rights, FileShare share, int bufferSize, FileOptions options, FileSecurity fileSecurity)
+        {
+            if (fileInfo == null)
+            {
+                throw new ArgumentNullException(nameof(fileInfo));
+            }
+
+            if (fileSecurity == null)
+            {
+                throw new ArgumentNullException(nameof(fileSecurity));
+            }
+
+            // don't include inheritable in our bounds check for share
+            FileShare tempshare = share & ~FileShare.Inheritable;
+
+            string badArg = null;
+            if (mode < FileMode.CreateNew || mode > FileMode.Append)
+            {
+                badArg = nameof(mode);
+            }
+            else if (tempshare < FileShare.None || tempshare > (FileShare.ReadWrite | FileShare.Delete))
+            {
+                badArg = nameof(share);
+            }
+            if (badArg != null)
+            {
+                throw new ArgumentOutOfRangeException(badArg, SR.ArgumentOutOfRange_Enum);
+            }
```

Comment:
This seems like unnecessary complication to avoid a throw site in a method that's not particularly hot path.  Can we instead just do:
```C#
if (mode < FileMode.CreateNew || mode > FileMode.Append)
{
    throw new ArgumentOutOfRangeException(nameof(mode), SR.ArgumentOutOfRange_Enum);
}

if (tempshare < FileShare.None || tempshare > (FileShare.ReadWrite | FileShare.Delete))
{
    throw new ArgumentOutOfRangeException(nameof(tempshare), SR.ArgumentOutOfRange_Enum);
}
```
It'd be easier to follow.

Reply:
This is copy&paste from FileStream. It would be nice to fix it in both places.

Reply:
Done.